docs/modules/user-guide/partials/proc_adding-a-registry-schema.adoc

```diff
@@ -0,0 +1,423 @@
+* tags: string[] - the tags associated to devfile
+* icon: string - the devfile icon
+* language: string - the project language that is used in the devfile
+* git: *git - the information of remote repositories
+
+.extraDevfileEntries.yaml sample
+
+====
+----
+schemaVersion: 1.0.0
+samples:
+  - name: nodejs-basic
+    displayName: Basic NodeJS
+    description: A simple Hello World Node.js application
+    icon: https://github.com/maysunfaisal/node-bulletin-board-2/blob/main/nodejs-icon.png
+    tags: ["NodeJS", "Express"]
+    projectType: nodejs
+    language: nodejs
+    git:
+      remotes:
+        origin: https://github.com/redhat-developer/devfile-sample.git
+  - name: code-with-quarkus
+    displayName: Basic Quarkus
+    description: A simple Hello World Java application using Quarkus
+    icon: .devfile/icon/quarkus.png
+    tags: ["Java", "Quarkus"]
+    projectType: quarkus
+    language: java
+    git:
+      remotes:
+        origin: https://github.com/elsony/devfile-sample-code-with-quarkus.git
+  - name: java-springboot-basic
+    displayName: Basic Spring Boot
+    description: A simple Hello World Java Spring Boot application using Maven
+    icon: .devfile/icon/spring-logo.png
+    tags: ["Java", "Spring"]
+    projectType: springboot
+    language: java
+    git:
+      remotes:
+        origin: https://github.com/elsony/devfile-sample-java-springboot-basic.git
+  - name: python-basic
+    displayName: Basic Python
+    description: A simple Hello World application using Python
+    icon: .devfile/icon/python.png
+    tags: ["Python"]
+    projectType: python
+    language: python
+    git:
+      remotes:
+        origin: https://github.com/elsony/devfile-sample-python-basic.git
+----
+====
+
+
+The file, index.json, contains the metadata of the stacks and samples in the registry. This metadata is also known as the devfile index schema. From the devfile index schema, you can get the stack and sample generic information such as: name, language, and resources.
+
+.Devfile index schema
+* name: string - the stack name
+* version: string - the stack version
+* attributes: map[string]apiext.JSON - map of implementation-dependant free-form YAML attributes
+* displayName: string - the display name of devfile
+* description: string - the description of devfile
+* type: DevfileType - the type of the devfile, currently supports stack and sample
+* tags: string[] - the tags associated to devfile
+* icon: string - the devfile icon
+* globalMemoryLimit: string - the devfile global memory limit
+* projectType: string - the project framework that is used in the devfile
+* language: string - the project language that is used in the devfile
+* links: map[string]string - links related to the devfile
+* resources: string[] - the file resources that compose a devfile stack.
+* starterProjects: string[] - the project templates that can be used in the devfile
+* git: *git - the information of remote repositories
+
+.Index.json sample
```

Comment:
This is a very, very large sample file.

Question for SMEs: is there a smaller, simpler file that we can use as an example? I ask because including all of this content here seems like a maintenance nightmare as things change in the future. Perhaps it would be best to create an explicit sample JSON file and link to it _directly_ rather than copying its contents into the documentation?

Just some questions worth thinking about as we continue -- I don't think these questions should directly stop this PR from being merged as it is now.

Reply:
We can shorten it. Something like:
```
[
  {
    "name": "java-maven",
    "version": "1.1.0",
    "displayName": "Maven Java",
    "description": "Upstream Maven and OpenJDK 11",
    "type": "stack",
    "tags": [
      "Java",
      "Maven"
    ],
    "projectType": "maven",
    "language": "java",
    "links": {
      "self": "devfile-catalog/java-maven:latest"
    },
    "resources": [
      "devfile.yaml"
    ],
    "starterProjects": [
      "springbootproject"
    ]
  },
  {
    "name": "java-openliberty",
    "version": "0.5.0",
    "displayName": "Open Liberty",
    "description": "Java application stack using Open Liberty runtime",
    "type": "stack",
    "projectType": "docker",
    "language": "java",
    "links": {
      "self": "devfile-catalog/java-openliberty:latest"
    },
    "resources": [
      "devfile.yaml"
    ],
    "starterProjects": [
      "user-app"
    ]
  }
]
```

Should be more than sufficient